From 9a27944eaa09552e02137105bcbd96b0e4895999 Mon Sep 17 00:00:00 2001
From: psathisx <pinayourx.sathish@intel.com>
Date: Fri, 21 Dec 2012 17:40:18 +0530
Subject: [PORT FROM MAIN] While trying to play WAV(PCM) 192kHz song (which is not supported),the next song playback does not start.

BZ: 68252

The Sampling rate check is added in the parser which will ensure that an error
is thrown if the sampling rate is unsupported much before the AudioFlinger
does the same.This fix will ensure that when the user tries to play an
unsupported format,the music app will either throw "The player doesn't support
this type of audio file" message or will jump to the next song and start
playing.

Change-Id: I974a9003ba30cbc486640814048131da3cdfb3d7
Orig-Change-Id: Ie0f63a5b29dc2881bd40e30a9dbaa24698fdb863
Signed-off-by: psathisx <pinayourx.sathish@intel.com>
Reviewed-on: http://android.intel.com:8080/83388
Reviewed-by: Sameullah, MazharX <mazharx.sameullah@intel.com>
Reviewed-by: Bhakte, GurudattaX <gurudattax.bhakte@intel.com>
Reviewed-by: Kandasamy, Muthukumar <muthukumar.kandasamy@intel.com>
Reviewed-by: Sikkandar D, Madar <madar.sikkandar.d@intel.com>
Tested-by: Gupta, ArvindX K <arvindx.k.gupta@intel.com>
Reviewed-by: P C, SreekanthX <sreekanthx.p.c@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libstagefright/WAVExtractor.cpp |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/WAVExtractor.cpp b/media/libstagefright/WAVExtractor.cpp
index 22af6fb..353686a 100644
--- a/media/libstagefright/WAVExtractor.cpp
+++ b/media/libstagefright/WAVExtractor.cpp
@@ -32,6 +32,10 @@
 
 #define CHANNEL_MASK_USE_CHANNEL_ORDER 0
 
+//Maximum input sampling rate that can be handled by the resampler.
+//This is a limitation of the resampler and not the WAV format.
+#define MAX_INPUT_SAMPLING_RATE 96000
+
 namespace android {
 
 enum {
@@ -206,7 +210,7 @@ status_t WAVExtractor::init() {
 
             mSampleRate = U32_LE_AT(&formatSpec[4]);
 
-            if (mSampleRate == 0) {
+            if (mSampleRate <= 0 || mSampleRate > MAX_INPUT_SAMPLING_RATE) {
                 return ERROR_MALFORMED;
             }
 
-- 
1.7.4.1

